### PR TITLE
Added colab notebook for state params

### PR DESCRIPTION
**Pull request by AdityaKane2001**
Regarding #1910 
Added "Open in Colab" button which points to a Colab notebook having executable code of the [Managing Parameters and State](https://flax.readthedocs.io/en/latest/howtos/state_params.html) guide.

## Checklist
- [ ] This PR fixes a minor issue (e.g.: typo or small bug) or improves the docs (you can dismiss the other
      checks if that's the case).
- [x] This change is discussed in a Github issue/
      [discussion](https://github.com/google/flax/discussions) (please add a
      link).
- [ ] The documentation and docstrings adhere to the
      [documentation guidelines](https://github.com/google/flax/blob/main/docs/README.md#how-to-write-code-documentation).
- [ ] This change includes necessary high-coverage tests.
      (No quality testing = no merge!)

### CODE REVIEW

**Comment by AdityaKane2001** [comment]
@marcvanzee 

Could you please take a look at this one? TIA

**Comment by marcvanzee** [comment]
Thanks a lot @AdityaKane2001!

Would it be possible to also include the Colab itself in this PR, and add it to `flax/docs/howtos/`? Then we will be able to update this if we need to change anything.

I also found that the Colab doesn't work, you need to change to dummy input to:

```py
dummy_input = jnp.ones((32, 5))
```

**Comment by AdityaKane2001** [comment]
@marcvanzee 

Could you please take a look and let me know it this is what you wanted?

File: docs/howtos/state_params.rst [hunk 1]
```diff
@@ -1,3 +1,6 @@
+.. image:: https://colab.research.google.com/assets/colab-badge.svg
+   :target: https://colab.research.google.com/gist/AdityaKane2001/cbd09c2fcb86163cb301076cb60ce200/state_params_howto.ipynb
```
Comment (marcvanzee) [inline docs/howtos/state_params.rst:2]: Please point to the Colab in this repo

File: docs/howtos/state_params_howto.ipynb [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        "model = MLP(hidden_size=10, out_size=1)\n",
+        "variables = model.init(random.PRNGKey(0), dummy_input)\n",
+        "# Split state and params (which are updated by optimizer).\n",
+        "state, params = variables.pop('params')\n",
+        "del variables  # Delete variables to avoid wasting resources\n",
+        "tx = optax.sgd(learning_rate=0.02)\n",
+        "opt_state = tx.init(params)\n",
+        "\n",
+        "for _ in range(num_epochs):\n",
+        "  opt_state, params, state, loss = update_step(\n",
```
Comment (marcvanzee) [inline docs/howtos/state_params_howto.ipynb:279]: It would be nice to print something. maybe print the loss after each epoch?

**Review by marcvanzee — CHANGES_REQUESTED** [review]
Thanks! Just a few small comments.

**Comment by marcvanzee** [comment]
It seems you cannot use `%%capture` in Colabs (it doesn't work with Sphinx). Could you remove that from the Colab as well? Right now it breaks our build.

File: docs/howtos/state_params_howto.ipynb (hunk 2 above)
Comment (AdityaKane2001) [reply docs/howtos/state_params_howto.ipynb:279]: Sure, I didn't add that earlier because I thought it would be inconsistent with the docs.

**Comment by AdityaKane2001** [comment]
Should I keep `#@title` or should I remove that as well?

**Comment by marcvanzee** [comment]
> Should I keep `#@title` or should I remove that as well?

I think you can keep that, since it is in a comment.

File: docs/howtos/state_params.rst [hunk 3]
```diff
@@ -1,3 +1,6 @@
+.. image:: https://colab.research.google.com/assets/colab-badge.svg
+   :target: https://github.com/google/flax/blob/main/docs/howtos/state_params_howto.ipynb
```
Comment (marcvanzee) [inline docs/howtos/state_params.rst:2]:
```suggestion
   :target: state_params_howto.ipynb
```

File: docs/howtos/state_params.rst [hunk 4]
```diff
@@ -1,3 +1,6 @@
+.. image:: https://colab.research.google.com/assets/colab-badge.svg
+   :target: state_params_howto.ipynb
```
Comment (marcvanzee) [inline docs/howtos/state_params.rst:2]:
```suggestion
   :target: https://github.com/google/flax/blob/main/docs/howtos/state_params_howto.ipynb
```

**Review by marcvanzee — APPROVED** [review]
Looks great, thanks a lot!

**Comment by AdityaKane2001** [comment]
Thanks for the merge!
